pallets/vesting-reward/src/lib.rs

```diff
@@ -0,0 +1,295 @@
+        DirectReward = BalanceOf<T>,
+        VestedReward = BalanceOf<T>,
+        BlockNumber = <T as frame_system::Config>::BlockNumber,
+    {
+        /// Event emitted when a reward claim was processed successfully.
+        RewardClaimed(AccountId, DirectReward, VestedReward),
+        /// Event triggered when the reward module is ready to reward contributors
+        /// [vesting_start, vesting_period, conversion_rate, direct_payout_ratio]
+        RewardModuleInitialized(BlockNumber, BlockNumber, Perbill, Perbill),
+        /// Direct payout ratio for contributors has been updated
+        UpdateDirectPayoutRatio(Perbill),
+        /// Conversion rate from relay to native token has been updated
+        UpdateConversionRate(Perbill),
+        /// Vesting period has been updated
+        UpdateVestingPeriod(BlockNumber),
+        /// Start of vesting has been updated
+        UpadteVestingStart(BlockNumber),
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Config> as Reward {
+        /// The conversion rate between relay chain and native chain balances.
+        pub ConversionRate get(fn conversion_rate) build(|config: &GenesisConfig| Perbill::from_percent(config.conversion)): Perbill;
+        /// Which ratio of the rewards are payed directly. The rest is transferred via a vesting schedule.
+        pub DirectPayoutRatio get(fn direct_payout_ratio) build(|config: &GenesisConfig| Perbill::from_percent(config.direct_payout)): Perbill;
+        /// Over which period are the contributions vested.
+        pub VestingPeriod get(fn vesting_period) : T::BlockNumber;
+        /// At which block number does the vesting start .
+        pub VestingStart get(fn vesting_start) : T::BlockNumber;
+    } add_extra_genesis {
+        config(direct_payout) : u32;
+        config(conversion) : u32;
+    }
+}
+
+decl_error! {
+	pub enum Error for Module<T: Config> {
+        /// Not enough funds in the pot for paying a reward
+        NotEnoughFunds,
+        /// Start of vesting period is in the past.
+        ElapsedTime,
+    }
+}
+
+
+decl_module! {
+    pub struct Module<T: Config> for enum Call where origin: <T as frame_system::Config>::Origin {
+        // Activate errors
+        type Error = Error<T>;
+
+        const MODULE_ACCOUNT_ID: <T as frame_system::Config>::AccountId = MODULE_ID.into_account();
+
+        // Activate events
+        fn deposit_event() = default;
+
+        #[weight = <T as Config>::WeightInfo::set_vesting_start()]
+        fn set_vesting_start(origin, start: T::BlockNumber) -> DispatchResult {
+            T::AdminOrigin::try_origin(origin)
+                .map(|_| ())
+                .or_else(ensure_root)?;
+
+            ensure!(
+                start >= <frame_system::Module<T>>::block_number(),
+                Error::<T>::ElapsedTime
+            );
+            <VestingStart<T>>::put(start);
+
+            Self::deposit_event(RawEvent::UpadteVestingStart(start));
```

Comment:
```suggestion
            Self::deposit_event(RawEvent::UpdateVestingStart(start));
```

Reply:
Noted. Will be done by the end of the week.